README.md

```diff
@@ -17,16 +17,16 @@ The HPack library has a simple interface. You will need two functions:
 
 ### Decoding
 ```elixir
-{:ok, ctx} = HPack.Table.start_link(1000)
-HPack.decode(<< 0x82 >>, ctx)
-# => [{":method", "GET"}]
+ctx = HPack.Table.new(1_000)
+{:ok, table, headers} = HPack.decode(ctx, (<< 0x82 >>)
```

Comment:
You already mentioned 

> It also cleans up the APIs to use the context as first argument which I think makes it more consistent and intuitive.

I'd like to better understand your reasoning here because I'm quite reluctant to pass the table first because I would not consider this the _subject_ of the function.

Reply:
There is not a lot of reasoning here, this is mostly a de facto standard in Elixir standard library and outside of it.
All functions operate on the table, operations without a table make no sense. It also makes all functions behave consistently, which is less surprising to the user and easier to use without needing to constantly lookup the docs.

Reply:
I also enables some pipeability in the APIs, I've updated the readme accordingly.

Reply:
Could you share some of that code, where you pipe in the ctx/table rather than the binary/headers? When I initially wrote the lib I thought about the order quite a bit. Looking at modules like Plug I realize many modules pass the state in first.

One nitpick, please do not pass scalar values with a pipe e.g.

```elixir
1000
|> Table.new
```

I always find this overkill.

Then I'll give it a full go, but concepts and major points are addressed :)

Reply:
Sure, I removed the integer pipes.

You can see some code I wrote with a composable API where you can pass the request into functions and build your request.

https://github.com/lucacorti/ankh/blob/master/lib/ankh/http/request.ex

Waiting for more feedback then 👍

Reply:
@lucacorti I was more looking for an example where you (or anyone) does something like this:

```elixir
HPack.Table.new(1_000)
|> HPack.encode([{":method", "GET"}])
```

Reply:
Personally I would do: 1_000 |> a() |> b(), I use that kind of pattern all the time, also piping into case statements.

e.g. https://github.com/lucacorti/ankh/blob/master/lib/ankh/http2.ex#L404

If you are referring specifically to HPack code, I'm not sure how others are using it, but I would use it like that.

Reply:
I was asking about examples of the argument order. For the examples I'd like to stick to the version we currently have.

After reading through your code I see you also do pipe in the hbf and pass the ctx/table as an argument:

https://github.com/lucacorti/ankh/blob/a4149dbcef699c5b5afdcaeb6f071d2ebbc0d99a/lib/ankh/http2.ex#L346-L347

IDK how many consumers we have for HPACK, but that's even for a major release such a PITA to change, in existing applications, that I like to stick to

```elixir
@spec encode(headers(), Table.t())
```
_(same for decode)_

for the time being, unless there is a real need (besides esthetics) to change. Consumers can always opt to create their own facade to HPACK, putting the args in whatever order they prefer.

Once that is addressed I'm happy to merge and release V3 :)

Reply:
I've reverted the argument order changes.